### PR TITLE
KAFKA-4642: Improve test coverage of ProcessorStateManager

### PR DESCRIPTION
**Pull request by dguy**
Most of the exception paths weren't covered. Now they are.

### CODE REVIEW

**Comment by dguy** [comment]
@enothereska @mjsax @guozhangwang

**Comment by enothereska** [comment]
LGTM

File: streams/src/test/java/org/apache/kafka/streams/processor/internals/ProcessorStateManagerTest.java [hunk 1]
```diff
@@ -520,4 +532,217 @@ public void shouldNotWriteCheckpointsIfAckeOffsetsIsNull() throws Exception {
         assertFalse(checkpointFile.exists());
     }
 
+    @Test
```
Comment (guozhangwang) [inline streams/src/test/java/org/apache/kafka/streams/processor/internals/ProcessorStateManagerTest.java:535]: We could use `@Test(expected = LockException.class), ditto below.

Comment (dguy) [reply streams/src/test/java/org/apache/kafka/streams/processor/internals/ProcessorStateManagerTest.java:535]: I prefer not to use that annotation in tests with multiple lines that aren't just doing something very simple, i.e., without side-effects. I want to make sure that the exception is thrown from the path i'm testing and not some other path that happens to be in the test. In this case i also want to release the lock to clean up.

**Review by guozhangwang — APPROVED** [review]

**Comment by guozhangwang** [comment]
Merged to trunk.

**Pull request closed without merging**